kbi/0026/index.rst

```diff
@@ -0,0 +1,53 @@
+.. index::
+   single: configurations; shared repository
+
+KBI0026: Passing configurations
+===============================
+
+:authors: Adina Wagner <adina.wagner@t-online.de>
+:discussion: <link>
+:keywords: configurations, shared repository clone
+:software-versions: datalad_0.19.1, datalad-next_1.0.0b3, git_2.39.2
+
+There are a variety of ways to configure datasets or DataLad operations.
+This KBI documents relevant resources, and highlights a few special cases and relevant recent developments about them.
+
+Existing resources
+------------------
+
+* A good introduction to the topic of configurations is contained in the `DataLad Handbook`_.
+* An overview of Git-specific configurations can be found by running ``git help --config``, or in the `Git documentation`_.
+* An overview of DataLad-specific configurations is in the `technical DataLad documentation`_.
+* Specific documentation and examples for the ``datalad configuration`` command are in its `manpage`_. This command can be used to set, unset, or query configurations, and complements the capabilities of ``git config`` with, among other things, additional scopes or recursive operations.
+
+.. _DataLad Handbook: https://handbook.datalad.org/en/latest/basics/basics-configuration.html
+.. _Git documentation: https://git-scm.com/docs/git-config#_variables
+.. _technical DataLad documentation: http://docs.datalad.org/en/stable/config.html
+.. _manpage: http://docs.datalad.org/en/stable/generated/man/datalad-configuration.html
+
+
+Recent developments
+-------------------
+
+The ``-c/--configuration`` flag of the ``datalad`` main command displays a number of shortcomings for specific applications.
+One of them is that configurations provided this way do not make it to the target process in every situation.
+For example, an attempt to override the user name (``user.name`` Git config) via ``datalad -c user.name=someoneelse save`` would be unsuccessful, and datalad or its sub-processes continues to use the user name configured in the applicable ``.config`` file.
+This shortcoming was `fixed`_ in May 2023.
+At the time of writing, the fix can be enabled with an installation of `datalad-next`_ and the configuration ``datalad.extensions.load next``, as detailed in the projects documentation.
+
+The alternative workaround requires the use of Git configurations in environment variables.
```

Comment:
It is not obvious to me how this line connects the following paragraph to the previous one. To what specific issue is the use of  Git configurations in environment variables an "alternative workaround"? I read it as "the following is an alternative to using datalad-next to solve the issue of not being able to override the git user name", but what then follows is a description and solution of a different issue (cloning from a shared repo). I'm guessing the issues and workarounds are supposed to be interpreted on a more abstract level, and examples are given to make it more tangible, but I think some rephrasing could be done to make this message clearer.

Reply:
thanks for the critical review, I really appreciate that! I have pushed a restructuring, could you give it a read when its convenient for you and see if that improves coherence?

Reply:
Restructured content makes it all much clearer, thanks!